### PR TITLE
Nav bar

### PR DESCRIPTION
**Pull request by matt-xia**
most of functionality complete except for logging in/out, opening settings modal and searching

### CODE REVIEW

File: flight-tracker-client/src/components/NavBar.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+function NavBar(props:{
+    changeTheme:()  => void,
+        currTheme:boolean
+    }) {
+    const [searchInput, setSearchInput] = useState<string>()
+    const [username, setUsername] = useState<string>()
+    const [password, setPassword] = useState<string>()
+    const [loggedIn, setLoggedIn] = useState<boolean>(true)
+    const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
```
Comment (J-Beswick) [inline flight-tracker-client/src/components/NavBar.tsx]: React.useState ?

**Review by workgiter — COMMENTED** [review]
Seems good. Could've divided the full NavBar functionality across multiple sub components.

File: flight-tracker-client/src/App.tsx [hunk 2]
```diff
@@ -1,34 +1,37 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import './App.css'
+import { createTheme, CssBaseline, ThemeProvider } from "@mui/material";
+import { useState, createContext, useMemo } from "react";
+import NavBar from "./components/NavBar";
+
+export const ColourModeContext = createContext({ toggleColourMode: () => {} });
```
Comment (J-Beswick) [inline flight-tracker-client/src/App.tsx:5]: toggleColourMode: () => void
is that better?

File: flight-tracker-client/src/components/NavBar.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+import MenuIcon from "@mui/icons-material/Menu";
+import "../components.css";
+import { ColourModeContext } from "../App";
+
+function NavBar() {
+  const [searchInput, setSearchInput] = useState<string>();
+  const [username, setUsername] = useState<string>();
+  const [password, setPassword] = useState<string>();
+  const [loggedIn, setLoggedIn] = useState<boolean>(true);
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
```
Comment (J-Beswick) [inline flight-tracker-client/src/components/NavBar.tsx]: react.usestate()

File: flight-tracker-client/src/components/NavBar.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  const loginUser = () => {
+    setLoggedIn((prev) => !prev);
+  };
+
+  const logoutUser = () => {
+    setLoggedIn((prev) => !prev);
+  };
+
+  return (
+    <Box sx={{ flexGrow: 1 }}>
```
Comment (workgiter) [inline flight-tracker-client/src/components/NavBar.tsx:47]: Could split functionality into more components (Search Box, AccountBox, etc) for the future thought it's not as useful with the code at it's current level of complexity.

**Pull request merged**